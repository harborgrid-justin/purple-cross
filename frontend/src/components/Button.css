/* Button Component Styles */

.btn {
  font-family: inherit;
  font-weight: 500;
  border: none;
  border-radius: var(--radius-md, 0.5rem);
  cursor: pointer;
  transition: all 0.2s ease-in-out;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  gap: var(--spacing-xs, 0.25rem);
  text-decoration: none;
  line-height: 1.5;
}

.btn:focus-visible {
  outline: 2px solid var(--primary-color, #7c3aed);
  outline-offset: 2px;
}

.btn:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

/* Button Sizes */
.btn-sm {
  padding: var(--spacing-xs, 0.25rem) var(--spacing-sm, 0.5rem);
  font-size: 0.875rem;
  min-height: 32px;
}

.btn-md {
  padding: var(--spacing-sm, 0.5rem) var(--spacing-lg, 1.5rem);
  font-size: 1rem;
  min-height: 44px; /* WCAG 2.1 AA touch target */
}

.btn-lg {
  padding: var(--spacing-md, 1rem) var(--spacing-xl, 2rem);
  font-size: 1.125rem;
  min-height: 48px;
}

/* Button Variants */
.btn-primary {
  background-color: var(--primary-color, #7c3aed);
  color: white;
}

.btn-primary:hover:not(:disabled) {
  background-color: var(--primary-dark, #6d28d9);
}

.btn-primary:active:not(:disabled) {
  background-color: #5b21b6;
}

.btn-secondary {
  background-color: var(--bg-primary, #ffffff);
  color: var(--text-primary, #1f2937);
  border: 1px solid var(--border-color, #e5e7eb);
}

.btn-secondary:hover:not(:disabled) {
  background-color: var(--bg-secondary, #f9fafb);
  border-color: var(--text-secondary, #6b7280);
}

.btn-danger {
  background-color: var(--danger-color, #ef4444);
  color: white;
}

.btn-danger:hover:not(:disabled) {
  background-color: #dc2626;
}

.btn-danger:active:not(:disabled) {
  background-color: #b91c1c;
}

.btn-success {
  background-color: var(--secondary-color, #10b981);
  color: white;
}

.btn-success:hover:not(:disabled) {
  background-color: #059669;
}

.btn-success:active:not(:disabled) {
  background-color: #047857;
}

/* Button Modifiers */
.btn-full-width {
  width: 100%;
}

.btn-loading {
  opacity: 0.7;
  cursor: wait;
  pointer-events: none;
}

/* Spinner Animation */
.spinner {
  width: 1em;
  height: 1em;
  border: 2px solid currentColor;
  border-top-color: transparent;
  border-radius: 50%;
  animation: spin 0.6s linear infinite;
}

@keyframes spin {
  to {
    transform: rotate(360deg);
  }
}

/* Reduced Motion Support */
@media (prefers-reduced-motion: reduce) {
  .spinner {
    animation-duration: 1.2s;
  }
}
