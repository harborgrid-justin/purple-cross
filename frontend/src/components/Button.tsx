/**
 * WF-COMP-XXX | Button.tsx - Button
 * Purpose: React component for Button functionality
 * Dependencies: react
 * Last Updated: 2025-10-23 | File Type: .tsx
 */

import { ButtonHTMLAttributes, ReactNode } from 'react';
import clsx from 'clsx';

/**
 * Props for the Button component.
 *
 * @interface ButtonProps
 * @extends {ButtonHTMLAttributes<HTMLButtonElement>}
 *
 * @property {ReactNode} children - The content to display inside the button
 * @property {'primary' | 'secondary' | 'danger' | 'success'} [variant='primary'] - Visual style variant
 * @property {'sm' | 'md' | 'lg'} [size='md'] - Button size
 * @property {boolean} [fullWidth=false] - Whether button should take full width of container
 * @property {boolean} [loading=false] - Loading state - shows spinner and disables button
 *
 * @example
 * // Primary button
 * <Button variant="primary" onClick={handleClick}>
 *   Save Changes
 * </Button>
 *
 * @example
 * // Loading state
 * <Button loading={isSubmitting}>
 *   Submit Form
 * </Button>
 */
interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  children: ReactNode;
  variant?: 'primary' | 'secondary' | 'danger' | 'success';
  size?: 'sm' | 'md' | 'lg';
  fullWidth?: boolean;
  loading?: boolean;
}

/**
 * Button component with multiple variants, sizes, and loading state support.
 *
 * @component
 * @param {ButtonProps} props - The component props
 * @returns {JSX.Element} Rendered button element
 *
 * @example
 * // Basic usage
 * <Button onClick={handleSave}>Save</Button>
 *
 * @example
 * // With variant and size
 * <Button variant="danger" size="lg" onClick={handleDelete}>
 *   Delete Account
 * </Button>
 *
 * @example
 * // Full width button
 * <Button fullWidth variant="primary" onClick={handleSubmit}>
 *   Submit Form
 * </Button>
 *
 * @example
 * // Loading state
 * <Button loading={isProcessing} disabled={!isValid}>
 *   Process Payment
 * </Button>
 *
 * @remarks
 * - Extends all native button HTML attributes
 * - Automatically disables button when loading={true}
 * - Uses clsx for conditional className composition
 * - Loading state shows spinner with "Loading..." text
 * - Supports custom className for additional styling
 */
const Button = ({
  children,
  variant = 'primary',
  size = 'md',
  fullWidth = false,
  loading = false,
  disabled,
  className,
  ...props
}: ButtonProps) => {
  return (
    <button
      className={clsx(
        'btn',
        `btn-${variant}`,
        `btn-${size}`,
        {
          'btn-full-width': fullWidth,
          'btn-loading': loading,
        },
        className
      )}
      disabled={disabled || loading}
      {...props}
    >
      {loading ? (
        <>
          <span className="spinner" aria-hidden="true"></span>
          <span>Loading...</span>
        </>
      ) : (
        children
      )}
    </button>
  );
};

export default Button;
