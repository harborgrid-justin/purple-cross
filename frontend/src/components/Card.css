/* Card Component Styles */

.card {
  background-color: var(--bg-primary, #ffffff);
  border-radius: var(--radius-lg, 0.75rem);
  box-shadow: var(--shadow, 0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06));
  overflow: hidden;
  transition: box-shadow 0.2s ease-in-out;
}

.card:hover {
  box-shadow: var(--shadow-lg, 0 10px 15px -3px rgba(0, 0, 0, 0.1), 0 4px 6px -2px rgba(0, 0, 0, 0.05));
}

/* Card Header */
.card-header {
  padding: var(--spacing-lg, 1.5rem);
  border-bottom: 1px solid var(--border-color, #e5e7eb);
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.card-header-content {
  flex: 1;
  min-width: 0; /* Allows text truncation */
}

.card-title {
  margin: 0;
  font-size: 1.25rem;
  font-weight: 600;
  color: var(--text-primary, #1f2937);
  line-height: 1.5;
}

.card-subtitle {
  margin: 0.25rem 0 0;
  font-size: 0.875rem;
  color: var(--text-secondary, #6b7280);
  line-height: 1.5;
}

.card-actions {
  display: flex;
  align-items: center;
  gap: var(--spacing-sm, 0.5rem);
  margin-left: var(--spacing-md, 1rem);
}

/* Card Body */
.card-body {
  padding: var(--spacing-lg, 1.5rem);
}

.card-body-no-padding {
  padding: 0;
}

/* Card Footer */
.card-footer {
  padding: var(--spacing-lg, 1.5rem);
  border-top: 1px solid var(--border-color, #e5e7eb);
  background-color: var(--bg-secondary, #f9fafb);
}

/* Card Variants */
.card-bordered {
  border: 1px solid var(--border-color, #e5e7eb);
  box-shadow: none;
}

.card-flat {
  box-shadow: none;
}

.card-elevated {
  box-shadow: var(--shadow-lg, 0 10px 15px -3px rgba(0, 0, 0, 0.1), 0 4px 6px -2px rgba(0, 0, 0, 0.05));
}

/* Interactive Card */
.card-interactive {
  cursor: pointer;
  transition: transform 0.2s ease-in-out, box-shadow 0.2s ease-in-out;
}

.card-interactive:hover {
  transform: translateY(-2px);
}

.card-interactive:active {
  transform: translateY(0);
}

/* Responsive */
@media (max-width: 768px) {
  .card-header,
  .card-body,
  .card-footer {
    padding: var(--spacing-md, 1rem);
  }

  .card-title {
    font-size: 1.125rem;
  }
}
