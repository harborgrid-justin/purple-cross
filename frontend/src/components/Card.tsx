/**
 * WF-COMP-XXX | Card.tsx - Card
 * Purpose: React component for Card functionality
 * Dependencies: react
 * Last Updated: 2025-10-23 | File Type: .tsx
 */

import { ReactNode } from 'react';
import clsx from 'clsx';

interface CardProps {
  children: ReactNode;
  className?: string;
  title?: string;
  subtitle?: string;
  actions?: ReactNode;
  noPadding?: boolean;
}

const Card = ({ children, className, title, subtitle, actions, noPadding = false }: CardProps) => {
  return (
    <div className={clsx('card', className)}>
      {(title || subtitle || actions) && (
        <div className="card-header">
          <div className="card-header-content">
            {title && <h2 className="card-title">{title}</h2>}
            {subtitle && <p className="card-subtitle">{subtitle}</p>}
          </div>
          {actions && <div className="card-actions">{actions}</div>}
        </div>
      )}
      <div className={clsx('card-body', { 'card-body-no-padding': noPadding })}>{children}</div>
    </div>
  );
};

export default Card;
