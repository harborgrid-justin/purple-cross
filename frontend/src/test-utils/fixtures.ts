/**
 * WF-COMP-XXX | fixtures.ts - Test Fixtures
 * Purpose: Mock data fixtures for testing
 * Dependencies: None
 * Last Updated: 2025-10-24 | File Type: .ts
 */

/**
 * Mock patient data
 */
export const mockPatient = {
  id: '1',
  name: 'Buddy',
  species: 'Dog',
  breed: 'Golden Retriever',
  dateOfBirth: '2020-01-15',
  gender: 'Male',
  color: 'Golden',
  weight: 32.5,
  microchipNumber: '123456789012345',
  status: 'active',
  clientId: '1',
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

export const mockPatients = [
  mockPatient,
  {
    id: '2',
    name: 'Whiskers',
    species: 'Cat',
    breed: 'Siamese',
    dateOfBirth: '2019-05-20',
    gender: 'Female',
    color: 'Cream',
    weight: 4.2,
    microchipNumber: '987654321098765',
    status: 'active',
    clientId: '1',
    createdAt: '2025-01-01T00:00:00Z',
    updatedAt: '2025-01-01T00:00:00Z',
  },
  {
    id: '3',
    name: 'Max',
    species: 'Dog',
    breed: 'German Shepherd',
    dateOfBirth: '2018-03-10',
    gender: 'Male',
    color: 'Black and Tan',
    weight: 38.0,
    status: 'inactive',
    clientId: '2',
    createdAt: '2025-01-01T00:00:00Z',
    updatedAt: '2025-01-01T00:00:00Z',
  },
];

/**
 * Mock client data
 */
export const mockClient = {
  id: '1',
  firstName: 'John',
  lastName: 'Doe',
  email: 'john.doe@example.com',
  phone: '555-0100',
  address: '123 Main St',
  city: 'Springfield',
  state: 'IL',
  zipCode: '62701',
  status: 'active',
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

export const mockClients = [
  mockClient,
  {
    id: '2',
    firstName: 'Jane',
    lastName: 'Smith',
    email: 'jane.smith@example.com',
    phone: '555-0200',
    address: '456 Oak Ave',
    city: 'Springfield',
    state: 'IL',
    zipCode: '62702',
    status: 'active',
    createdAt: '2025-01-01T00:00:00Z',
    updatedAt: '2025-01-01T00:00:00Z',
  },
];

/**
 * Mock appointment data
 */
export const mockAppointment = {
  id: '1',
  patientId: '1',
  clientId: '1',
  staffId: '1',
  appointmentType: 'checkup',
  startTime: '2025-10-25T10:00:00Z',
  endTime: '2025-10-25T10:30:00Z',
  status: 'scheduled',
  notes: 'Annual wellness exam',
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

export const mockAppointments = [
  mockAppointment,
  {
    id: '2',
    patientId: '2',
    clientId: '1',
    staffId: '1',
    appointmentType: 'surgery',
    startTime: '2025-10-25T14:00:00Z',
    endTime: '2025-10-25T15:00:00Z',
    status: 'confirmed',
    notes: 'Spay procedure',
    createdAt: '2025-01-01T00:00:00Z',
    updatedAt: '2025-01-01T00:00:00Z',
  },
];

/**
 * Mock medical record data
 */
export const mockMedicalRecord = {
  id: '1',
  patientId: '1',
  staffId: '1',
  visitDate: '2025-10-20',
  chiefComplaint: 'Annual checkup',
  diagnosis: 'Healthy',
  treatment: 'Vaccinations administered',
  prescriptions: [],
  labTests: [],
  status: 'completed',
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

export const mockMedicalRecords = [
  mockMedicalRecord,
  {
    id: '2',
    patientId: '1',
    staffId: '1',
    visitDate: '2025-09-15',
    chiefComplaint: 'Ear infection',
    diagnosis: 'Otitis externa',
    treatment: 'Antibiotic ear drops',
    prescriptions: ['1'],
    labTests: [],
    status: 'completed',
    createdAt: '2025-01-01T00:00:00Z',
    updatedAt: '2025-01-01T00:00:00Z',
  },
];

/**
 * Mock prescription data
 */
export const mockPrescription = {
  id: '1',
  patientId: '1',
  medicationName: 'Amoxicillin',
  dosage: '250mg',
  frequency: 'Twice daily',
  duration: '10 days',
  instructions: 'Give with food',
  prescribedBy: '1',
  status: 'active',
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

/**
 * Mock staff data
 */
export const mockStaff = {
  id: '1',
  firstName: 'Dr. Sarah',
  lastName: 'Johnson',
  email: 'sarah.johnson@purplecross.com',
  phone: '555-0300',
  role: 'veterinarian',
  specialty: 'General Practice',
  licenseNumber: 'VET-12345',
  status: 'active',
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

/**
 * Mock invoice data
 */
export const mockInvoice = {
  id: '1',
  clientId: '1',
  patientId: '1',
  invoiceNumber: 'INV-2025-001',
  invoiceDate: '2025-10-20',
  dueDate: '2025-11-20',
  subtotal: 150.0,
  tax: 12.0,
  total: 162.0,
  amountPaid: 0.0,
  status: 'unpaid',
  items: [
    {
      description: 'Annual wellness exam',
      quantity: 1,
      unitPrice: 100.0,
      total: 100.0,
    },
    {
      description: 'Rabies vaccination',
      quantity: 1,
      unitPrice: 50.0,
      total: 50.0,
    },
  ],
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

/**
 * Mock inventory item data
 */
export const mockInventoryItem = {
  id: '1',
  name: 'Amoxicillin 250mg',
  category: 'Medication',
  sku: 'MED-001',
  quantity: 100,
  reorderLevel: 20,
  unitPrice: 2.5,
  supplier: 'VetSupply Co',
  expirationDate: '2026-12-31',
  status: 'active',
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

/**
 * Mock lab test data
 */
export const mockLabTest = {
  id: '1',
  patientId: '1',
  testType: 'Blood panel',
  testDate: '2025-10-20',
  results: 'All values within normal range',
  status: 'completed',
  priority: 'routine',
  orderedBy: '1',
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

/**
 * Mock document data
 */
export const mockDocument = {
  id: '1',
  patientId: '1',
  clientId: '1',
  title: 'Vaccination Certificate',
  documentType: 'Certificate',
  fileName: 'vaccination-cert.pdf',
  fileSize: 245678,
  mimeType: 'application/pdf',
  uploadedBy: '1',
  status: 'signed',
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

/**
 * Mock user data
 */
export const mockUser = {
  id: '1',
  email: 'test@example.com',
  firstName: 'Test',
  lastName: 'User',
  role: 'veterinarian',
  token: 'mock-jwt-token',
};

/**
 * Mock API response
 */
export const mockApiResponse = <T>(data: T) => ({
  status: 'success',
  data,
});

/**
 * Mock API error response
 */
export const mockApiError = (message: string, statusCode = 400) => ({
  status: 'error',
  message,
  statusCode,
});

/**
 * Mock pagination metadata
 */
export const mockPagination = {
  page: 1,
  limit: 10,
  total: 100,
  totalPages: 10,
};

/**
 * Generate mock list with pagination
 */
export const mockPaginatedResponse = <T>(
  data: T[],
  page = 1,
  limit = 10,
  total?: number
) => ({
  status: 'success',
  data,
  pagination: {
    page,
    limit,
    total: total || data.length,
    totalPages: Math.ceil((total || data.length) / limit),
  },
});
