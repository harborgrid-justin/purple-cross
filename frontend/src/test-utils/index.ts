/**
 * WF-COMP-XXX | index.ts - Test Utils Exports
 * Purpose: Central export point for all test utilities
 * Dependencies: None
 * Last Updated: 2025-10-24 | File Type: .ts
 */

// Export custom render functions
export * from './render';

// Export test helpers
export * from './helpers';

// Export test fixtures
export * from './fixtures';

// Export MSW mocks
export * from './mocks/handlers';
export { server, mockServer } from './mocks/server';
