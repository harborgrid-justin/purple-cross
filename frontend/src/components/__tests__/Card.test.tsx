/**
 * WF-COMP-XXX | Card.test.tsx - Card Component Tests
 * Purpose: Comprehensive tests for Card component
 * Dependencies: @testing-library/react
 * Last Updated: 2025-10-24 | File Type: .tsx
 */

import { describe, it, expect } from 'vitest';
import { renderWithoutRouter } from '@/test-utils';
import Card from '../Card';
import Button from '../Button';

describe('Card Component', () => {
  describe('Rendering', () => {
    it('should render with children', () => {
      const { getByText } = renderWithoutRouter(
        <Card>
          <p>Card content</p>
        </Card>
      );

      expect(getByText('Card content')).toBeInTheDocument();
    });

    it('should render with default classes', () => {
      const { container } = renderWithoutRouter(
        <Card>
          <p>Content</p>
        </Card>
      );

      const card = container.firstChild;
      expect(card).toHaveClass('card');
    });

    it('should render card body', () => {
      const { container } = renderWithoutRouter(
        <Card>
          <p>Content</p>
        </Card>
      );

      const cardBody = container.querySelector('.card-body');
      expect(cardBody).toBeInTheDocument();
    });

    it('should render with custom className', () => {
      const { container } = renderWithoutRouter(
        <Card className="custom-card">
          <p>Content</p>
        </Card>
      );

      expect(container.firstChild).toHaveClass('card');
      expect(container.firstChild).toHaveClass('custom-card');
    });
  });

  describe('Title and Subtitle', () => {
    it('should render title', () => {
      const { getByText } = renderWithoutRouter(
        <Card title="Card Title">
          <p>Content</p>
        </Card>
      );

      expect(getByText('Card Title')).toBeInTheDocument();
    });

    it('should render title as h2 heading', () => {
      const { getByRole } = renderWithoutRouter(
        <Card title="Card Title">
          <p>Content</p>
        </Card>
      );

      const heading = getByRole('heading', { level: 2 });
      expect(heading).toHaveTextContent('Card Title');
      expect(heading).toHaveClass('card-title');
    });

    it('should render subtitle', () => {
      const { getByText } = renderWithoutRouter(
        <Card title="Title" subtitle="Card Subtitle">
          <p>Content</p>
        </Card>
      );

      expect(getByText('Card Subtitle')).toBeInTheDocument();
    });

    it('should render subtitle with correct class', () => {
      const { getByText } = renderWithoutRouter(
        <Card title="Title" subtitle="Subtitle">
          <p>Content</p>
        </Card>
      );

      const subtitle = getByText('Subtitle');
      expect(subtitle).toHaveClass('card-subtitle');
      expect(subtitle.tagName).toBe('P');
    });

    it('should render both title and subtitle', () => {
      const { getByText, getByRole } = renderWithoutRouter(
        <Card title="Main Title" subtitle="Helpful subtitle">
          <p>Content</p>
        </Card>
      );

      expect(getByRole('heading', { level: 2 })).toHaveTextContent('Main Title');
      expect(getByText('Helpful subtitle')).toBeInTheDocument();
    });

    it('should not render header when no title, subtitle, or actions', () => {
      const { container } = renderWithoutRouter(
        <Card>
          <p>Content only</p>
        </Card>
      );

      const header = container.querySelector('.card-header');
      expect(header).not.toBeInTheDocument();
    });

    it('should render header with only title', () => {
      const { container } = renderWithoutRouter(
        <Card title="Only Title">
          <p>Content</p>
        </Card>
      );

      const header = container.querySelector('.card-header');
      expect(header).toBeInTheDocument();
    });
  });

  describe('Actions', () => {
    it('should render actions in header', () => {
      const actions = <Button>Edit</Button>;
      const { getByRole } = renderWithoutRouter(
        <Card title="Title" actions={actions}>
          <p>Content</p>
        </Card>
      );

      expect(getByRole('button', { name: /edit/i })).toBeInTheDocument();
    });

    it('should render actions with correct class', () => {
      const actions = <Button>Save</Button>;
      const { container } = renderWithoutRouter(
        <Card title="Title" actions={actions}>
          <p>Content</p>
        </Card>
      );

      const actionsDiv = container.querySelector('.card-actions');
      expect(actionsDiv).toBeInTheDocument();
    });

    it('should render multiple action buttons', () => {
      const actions = (
        <>
          <Button>Edit</Button>
          <Button variant="danger">Delete</Button>
        </>
      );
      const { getByRole } = renderWithoutRouter(
        <Card title="Title" actions={actions}>
          <p>Content</p>
        </Card>
      );

      expect(getByRole('button', { name: /edit/i })).toBeInTheDocument();
      expect(getByRole('button', { name: /delete/i })).toBeInTheDocument();
    });

    it('should render header with only actions (no title)', () => {
      const actions = <Button>Close</Button>;
      const { container, getByRole } = renderWithoutRouter(
        <Card actions={actions}>
          <p>Content</p>
        </Card>
      );

      const header = container.querySelector('.card-header');
      expect(header).toBeInTheDocument();
      expect(getByRole('button', { name: /close/i })).toBeInTheDocument();
    });

    it('should position actions correctly in header', () => {
      const actions = <Button>Action</Button>;
      const { container } = renderWithoutRouter(
        <Card title="Title" actions={actions}>
          <p>Content</p>
        </Card>
      );

      const header = container.querySelector('.card-header');
      const actionsDiv = container.querySelector('.card-actions');

      expect(header).toContainElement(actionsDiv as HTMLElement);
    });
  });

  describe('No Padding Option', () => {
    it('should apply no-padding class when noPadding is true', () => {
      const { container } = renderWithoutRouter(
        <Card noPadding>
          <p>Content</p>
        </Card>
      );

      const cardBody = container.querySelector('.card-body');
      expect(cardBody).toHaveClass('card-body-no-padding');
    });

    it('should not apply no-padding class by default', () => {
      const { container } = renderWithoutRouter(
        <Card>
          <p>Content</p>
        </Card>
      );

      const cardBody = container.querySelector('.card-body');
      expect(cardBody).not.toHaveClass('card-body-no-padding');
    });

    it('should apply no-padding class when noPadding is false', () => {
      const { container } = renderWithoutRouter(
        <Card noPadding={false}>
          <p>Content</p>
        </Card>
      );

      const cardBody = container.querySelector('.card-body');
      expect(cardBody).not.toHaveClass('card-body-no-padding');
    });
  });

  describe('Content Rendering', () => {
    it('should render simple text content', () => {
      const { getByText } = renderWithoutRouter(
        <Card>
          <p>Simple text</p>
        </Card>
      );

      expect(getByText('Simple text')).toBeInTheDocument();
    });

    it('should render complex nested content', () => {
      const { getByText, getByRole } = renderWithoutRouter(
        <Card title="Complex Card">
          <div>
            <h3>Section Title</h3>
            <p>Paragraph text</p>
            <ul>
              <li>Item 1</li>
              <li>Item 2</li>
            </ul>
            <Button>Action</Button>
          </div>
        </Card>
      );

      expect(getByText('Section Title')).toBeInTheDocument();
      expect(getByText('Paragraph text')).toBeInTheDocument();
      expect(getByText('Item 1')).toBeInTheDocument();
      expect(getByRole('button', { name: /action/i })).toBeInTheDocument();
    });

    it('should render lists', () => {
      const { getByText } = renderWithoutRouter(
        <Card>
          <ul>
            <li>First item</li>
            <li>Second item</li>
            <li>Third item</li>
          </ul>
        </Card>
      );

      expect(getByText('First item')).toBeInTheDocument();
      expect(getByText('Second item')).toBeInTheDocument();
      expect(getByText('Third item')).toBeInTheDocument();
    });

    it('should render forms', () => {
      const { getByLabelText, getByRole } = renderWithoutRouter(
        <Card title="Form Card">
          <form>
            <label htmlFor="name">Name</label>
            <input id="name" type="text" />
            <Button type="submit">Submit</Button>
          </form>
        </Card>
      );

      expect(getByLabelText('Name')).toBeInTheDocument();
      expect(getByRole('button', { name: /submit/i })).toBeInTheDocument();
    });

    it('should render images', () => {
      const { getByAltText } = renderWithoutRouter(
        <Card>
          <img src="/test.jpg" alt="Test image" />
        </Card>
      );

      expect(getByAltText('Test image')).toBeInTheDocument();
    });
  });

  describe('Header Layout', () => {
    it('should render header content section', () => {
      const { container } = renderWithoutRouter(
        <Card title="Title" subtitle="Subtitle">
          <p>Content</p>
        </Card>
      );

      const headerContent = container.querySelector('.card-header-content');
      expect(headerContent).toBeInTheDocument();
    });

    it('should contain title and subtitle in header content', () => {
      const { container } = renderWithoutRouter(
        <Card title="Title" subtitle="Subtitle">
          <p>Content</p>
        </Card>
      );

      const headerContent = container.querySelector('.card-header-content');
      const title = container.querySelector('.card-title');
      const subtitle = container.querySelector('.card-subtitle');

      expect(headerContent).toContainElement(title);
      expect(headerContent).toContainElement(subtitle);
    });

    it('should render header with title and actions side by side', () => {
      const actions = <Button>Action</Button>;
      const { container } = renderWithoutRouter(
        <Card title="Title" actions={actions}>
          <p>Content</p>
        </Card>
      );

      const header = container.querySelector('.card-header');
      const headerContent = container.querySelector('.card-header-content');
      const cardActions = container.querySelector('.card-actions');

      expect(header).toContainElement(headerContent as HTMLElement);
      expect(header).toContainElement(cardActions as HTMLElement);
    });
  });

  describe('Accessibility', () => {
    it('should have proper semantic structure', () => {
      const { getByRole } = renderWithoutRouter(
        <Card title="Accessible Card">
          <p>Content</p>
        </Card>
      );

      expect(getByRole('heading', { level: 2 })).toHaveTextContent('Accessible Card');
    });

    it('should be navigable', () => {
      const { container } = renderWithoutRouter(
        <Card>
          <a href="/link">Link</a>
        </Card>
      );

      expect(container.querySelector('a')).toBeInTheDocument();
    });

    it('should support focus management with interactive elements', () => {
      const { getByRole } = renderWithoutRouter(
        <Card title="Interactive Card" actions={<Button>Action</Button>}>
          <Button>Content Button</Button>
        </Card>
      );

      const buttons = getByRole('button', { name: /action/i });
      expect(buttons).toBeInTheDocument();
    });
  });

  describe('Edge Cases', () => {
    it('should handle empty children', () => {
      const { container } = renderWithoutRouter(<Card />);

      const cardBody = container.querySelector('.card-body');
      expect(cardBody).toBeInTheDocument();
      expect(cardBody).toBeEmptyDOMElement();
    });

    it('should handle very long titles', () => {
      const longTitle = 'This is a very long title that might need to wrap across multiple lines in the card header';
      const { getByText } = renderWithoutRouter(
        <Card title={longTitle}>
          <p>Content</p>
        </Card>
      );

      expect(getByText(longTitle)).toBeInTheDocument();
    });

    it('should handle very long subtitles', () => {
      const longSubtitle = 'This is a very long subtitle with lots of explanatory text that provides context for the card content';
      const { getByText } = renderWithoutRouter(
        <Card title="Title" subtitle={longSubtitle}>
          <p>Content</p>
        </Card>
      );

      expect(getByText(longSubtitle)).toBeInTheDocument();
    });

    it('should handle null children gracefully', () => {
      const { container } = renderWithoutRouter(
        <Card title="Title">{null}</Card>
      );

      const cardBody = container.querySelector('.card-body');
      expect(cardBody).toBeInTheDocument();
    });

    it('should handle undefined title gracefully', () => {
      const { container } = renderWithoutRouter(
        <Card title={undefined}>
          <p>Content</p>
        </Card>
      );

      const header = container.querySelector('.card-header');
      expect(header).not.toBeInTheDocument();
    });
  });

  describe('Snapshot Tests', () => {
    it('should match snapshot for simple card', () => {
      const { container } = renderWithoutRouter(
        <Card>
          <p>Simple content</p>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for card with all features', () => {
      const { container } = renderWithoutRouter(
        <Card
          title="Full Featured Card"
          subtitle="With all props"
          actions={<Button>Action</Button>}
          className="custom-class"
        >
          <p>Card content</p>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for card with no padding', () => {
      const { container } = renderWithoutRouter(
        <Card noPadding>
          <p>No padding content</p>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
