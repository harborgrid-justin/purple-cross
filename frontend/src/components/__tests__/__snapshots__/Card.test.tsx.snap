// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card Component > Snapshot Tests > should match snapshot for card with all features 1`] = `
<div
  class="card custom-class"
>
  <div
    class="card-header"
  >
    <div
      class="card-header-content"
    >
      <h2
        class="card-title"
      >
        Full Featured Card
      </h2>
      <p
        class="card-subtitle"
      >
        With all props
      </p>
    </div>
    <div
      class="card-actions"
    >
      <button
        class="btn btn-primary btn-md"
      >
        Action
      </button>
    </div>
  </div>
  <div
    class="card-body"
  >
    <p>
      Card content
    </p>
  </div>
</div>
`;

exports[`Card Component > Snapshot Tests > should match snapshot for card with no padding 1`] = `
<div
  class="card"
>
  <div
    class="card-body card-body-no-padding"
  >
    <p>
      No padding content
    </p>
  </div>
</div>
`;

exports[`Card Component > Snapshot Tests > should match snapshot for simple card 1`] = `
<div
  class="card"
>
  <div
    class="card-body"
  >
    <p>
      Simple content
    </p>
  </div>
</div>
`;
